### PR TITLE
DNM: Test wheel jobs pre-release

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

With the 0.20.0 release pending this commit is just a throwaway test to
exercise the wheel build jobs prior to to the release. We made some
significant changes to the build system in this release so we should
sanity check that all the wheel jobs succeed before we tag the release
in order to avoid a firedrill post-release in case anything fails.

### CODE REVIEW

**Comment by mtreinish** [comment]
hmm, the gha wheel jobs didn't trigger. I might have to test that as a separate branch on the upstream repo instead of my fork. I'll resubmit after the azure wheel jobs finish

**Comment by mtreinish** [comment]
I pushed the gha jobs as a separate branch to trigger the jobs to run. You can see the status here: https://github.com/Qiskit/qiskit-terra/actions/runs/2071342031

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 name: Wheel Builds
 on:
   push:
-    tags:
-      - '*'
+    pull_request:
+      branches: ['main']
```
Comment (jakelishman) [inline .github/workflows/wheels.yml:6]: I think the trick is that `pull_request` isn't part of `push`, it's
```yaml
on:
  pull_request:
    ...
```

Comment (mtreinish) [reply .github/workflows/wheels.yml:6]: ah good catch. Well if I need to rerun https://github.com/Qiskit/qiskit-terra/actions/runs/2071342031 for some reason I'll fix this here and delete that branch. But hopefully everything will just work

**Comment by mtreinish** [comment]
The macOS arm64 job has been fixed by https://github.com/Qiskit/qiskit-terra/pull/7848/commits/e4eee94d02931747389a7522c22ce7f78060b83e and I integrated it into #7828 (see https://github.com/Qiskit/qiskit-terra/pull/7828/commits/033d0579c7ccf42decd0e991fd40a744cb7e3616 ). The linux and windows builds are failing because of a 32bit platform bug that @jakelishman is fixing in #7849 and we'll test and fix that there.The non-x86 linux jobs seem to be working fine (s390x is generally slow as the emulation of the isa is slower so it's still running but it looks like it's working). So I'm going to close this as it's served it's purpose

**Pull request closed without merging**